scripts/core/exp-usm-p2p.yml

```diff
@@ -0,0 +1,113 @@
+  - "It is not guaranteed that the commutation relations of the function arguments are identical for peer access and peer copies: For example for any given adapter, if, prior to the call, the peer device had access to device memory allocations on the command device, the peer device may still, following the function call, be able to copy data from the command device, but not access and write the same data on the command device."
+  - "Consult the appropriate adapter driver documentation for details of adapter specific behavior and native error codes that may be returned."
+params:
+    - type: $x_device_handle_t
+      name: commandDevice
+      desc: "[in] handle of the command device object"
+    - type: $x_device_handle_t
+      name: peerDevice
+      desc: "[in] handle of the peer device object"
+returns:
+    - $X_RESULT_SUCCESS
+    - $X_RESULT_ERROR_ADAPTER_SPECIFIC:
+      - "Returned if a native error code is returned by the adapter driver function disabling P2P. Consult the UR documentation for details on retrieving the corresponding native error code."
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Disable access to peer device memory"
+class: $xUsmP2P
+name: PeerAccessGetInfoExp
+ordinal: "0"
+analogue:
+    - "**cuDeviceGetP2PAttribute**"
+details:
+  - "Queries the peer access capabilities from the command device to the peer device according to the query `propName`."
+
+params:
+    - type: $x_device_handle_t
+      name: commandDevice
+      desc: "[in] handle of the command device object"
+    - type: $x_device_handle_t
+      name: peerDevice
+      desc: "[in] handle of the peer device object"
+
+    - type: $x_exp_peer_info_t
+      name: propName
+      desc: "[in] type of the info to retrieve"
+    - type: "size_t"
+      name: propSize
+      desc: |
+            [in] the number of bytes pointed to by pPropValue.
+    - type: "void*"
+      name: pPropValue
+      desc: |
+            [out][optional][typename(propName, propSize)] array of bytes holding the info.
+            If propSize is not equal to or greater than the real number of bytes needed to return the info
+            then the $X_RESULT_ERROR_INVALID_SIZE error is returned and pPropValue is not used.
+    - type: "size_t*"
+      name: pPropSizeRet
+      desc: |
+            [out][optional] pointer to the actual size in bytes of the queried propName.
+returns:
+    - $X_RESULT_SUCCESS
+    - $X_RESULT_ERROR_ADAPTER_SPECIFIC:
+      - "Returned if a native error code is returned by the adapter driver function querying the property. Consult the UR documentation for details on retrieving the corresponding native error code."
```

Comment:
Thanks, I've added these return types.
I think that it is important to have RESULT_ERROR_ADAPTER_SPECIFIC also for the general device/platform getinfo functions, so we can swap code like this that throws away backend specific errors by just calling an assert: https://github.com/intel/llvm/blob/1a6f7a51502bfe6bf177d7302cd1f08de4a8ec02/sycl/plugins/unified_runtime/ur/adapters/cuda/context.cpp#L70

See e.g. here to see the list of native errors that ` cuDeviceGetAttribute` can throw that are currently being thrown away by the cuda adapter: https://docs.nvidia.com/cuda/archive/10.2/cuda-driver-api/group__CUDA__DEVICE.html#group__CUDA__DEVICE_1g9c3e1414f0ad901d3278a4d6645fc266
Note also that any cu function can return errors from previous async cu driver calls. It is important to preserve all such native errors.
This is consistent with what the L0 plugin was doing in the past and is what we agreed to do generally here https://github.com/oneapi-src/unified-runtime/issues/500.
DPC++ already automatically retrieves the native backend error and passes it to a sycl::exception when `PI_ERROR_PLUGIN_SPECIFIC_ERROR` is returned from a PI call: we should check that it can do the same with `UR_RESULT_ERROR_ADAPTER_SPECIFIC` : actually I suspect that this is not the case and is something that we will need to fix.

Reply:
I've created #636 to track this 👍

Reply:
Yeah it was broke. I fixed it here: https://github.com/intel/llvm/pull/9975